microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/SubdeploymentHealthTest.java

```diff
@@ -0,0 +1,125 @@
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+/**
+ * Deployment with sub-deployments scenario.
+ */
+@Ignore("https://issues.redhat.com/browse/WFLY-12835")
+@RunWith(Arquillian.class)
+public class SubdeploymentHealthTest {
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    String healthURL;
+
+    @Before
+    public void composeHealthEndpointURL() {
+        healthURL = "http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort() + "/health";
+    }
+
+    @Deployment
+    public static EnterpriseArchive createDeployment() {
+
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class,
+                SubdeploymentHealthTest.class.getSimpleName() + ".ear");
+
+        ear.addAsModule(ShrinkWrap.create(WebArchive.class, "dep1.war")
+                .addClasses(SimplifiedHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml"));
+
+        ear.addAsModule(ShrinkWrap.create(JavaArchive.class, "dep2.jar")
+                .addClasses(LivenessHealthCheck.class)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"));
+
+        ear.setApplicationXML(new StringAsset("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                + "<application xmlns=\"http://xmlns.jcp.org/xml/ns/javaee\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/application_7.xsd\" version=\"7\">\n"
+                + "  <display-name>metrics</display-name>\n"
+                + "  <module>\n"
+                + "    <web>\n"
+                + "      <web-uri>dep1.war</web-uri>\n"
+                + "      <context-root>/dep</context-root>\n"
+                + "    </web>\n"
+                + "  </module>\n"
+                + "  <module>\n"
+                + "    <java>dep2.jar</java>\n"
+                + "  </module>\n"
+                + "</application>"));
+
+        return ear;
+    }
+
+    /**
+     * @tpTestDetails Customer deployment with sub-deployment scenario.
+     * @tpPassCrit Overall and the health check status is up.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    public void testHealthEndpoint() {
+        RestAssured.when().get(this.healthURL).then()
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(3), // BothHealthCheck contains both annotations: @Liveness and @Readiness
```

Comment:
BothHealthCheck -> SimplifiedHealthCheck or just remove the comment.

Reply:
Fixing...